openedx/core/djangoapps/content_tagging/rest_api/v1/tests/test_views.py

```diff
@@ -1044,6 +1045,121 @@ def _test_api_call(self, **kwargs) -> None:
             assert response.status_code == status.HTTP_404_NOT_FOUND
 
 
+@skip_unless_cms
+@ddt.ddt
+class TestTaxonomyUpdateOrg(TestTaxonomyObjectsMixin, APITestCase):
+    """
+    Test cases for updating orgs from taxonomies
+    """
+
+    def test_update_org(self) -> None:
+        """
+        Tests that taxonomy admin can add/remove orgs from a taxonomy
+        """
+        url = TAXONOMY_ORG_UPDATE_ORG_URL.format(pk=self.tA1.pk)
+        self.client.force_authenticate(user=self.staff)
+
+        response = self.client.put(url, {"orgs": [self.orgB.short_name, self.orgX.short_name]}, format="json")
+        assert response.status_code == status.HTTP_200_OK
+
+        # Check that the orgs were updated
+        taxonomy_orgs = TaxonomyOrg.objects.filter(taxonomy=self.tA1)
+        assert taxonomy_orgs.count() == 2
+        assert taxonomy_orgs[0].org == self.orgB
+        assert taxonomy_orgs[1].org == self.orgX
+
+    def test_update_all_org(self) -> None:
+        """
+        Tests that taxonomy admin can associate a taxonomy to all orgs
+        """
+        url = TAXONOMY_ORG_UPDATE_ORG_URL.format(pk=self.tA1.pk)
+        self.client.force_authenticate(user=self.staff)
+
+        response = self.client.put(url, {"all_orgs": True}, format="json")
+        assert response.status_code == status.HTTP_200_OK
+
+        # Check that the orgs were updated
+        taxonomy_orgs = TaxonomyOrg.objects.filter(taxonomy=self.tA1)
+        assert taxonomy_orgs.count() == 1
+        assert taxonomy_orgs[0].org is None
+
+    def test_update_no_org(self) -> None:
+        """
+        Tests that taxonomy admin can associate a taxonomy no orgs
+        """
+        url = TAXONOMY_ORG_UPDATE_ORG_URL.format(pk=self.tA1.pk)
+        self.client.force_authenticate(user=self.staff)
+
+        response = self.client.put(url, {"orgs": []}, format="json")
+
+        assert response.status_code == status.HTTP_200_OK
+
+        # Check that the orgs were updated
+        taxonomy_orgs = TaxonomyOrg.objects.filter(taxonomy=self.tA1)
+        assert taxonomy_orgs.count() == 0
+
+    @ddt.data(
+        (True, ["orgX"], "Using both all_orgs and orgs parameters should throw error"),
+        (False, None, "Using neither all_orgs or orgs parameter should throw error"),
```

Comment:
This case looks valid to me.. How do we toggle from `all_orgs = True` to `all_orgs = False`?

Reply:
You must pass the `all_orgs=False` and an empty orgs (`orgs=[]`) array. If you think that passing omitting `orgs` should have the same effect, let me know!

Personally, I prefer the explicitly `orgs=[]` way.

Reply:
Sure, as you wish.